warehouse/warehousecron/warehousecorn.go

```diff
@@ -0,0 +1,56 @@
+package warehousecron
+
+import (
+	"github.com/robfig/cron/v3"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	"strings"
+)
+
+type warehouseUploadCronJob struct {
+	warehouseUploadCron *cron.Cron
+	jobEntryID          cron.EntryID
+}
+
+var (
+	warehouseChan             chan bool
+	cronParser                cron.Parser
+	warehouseUploadCronJobMap map[string]warehouseUploadCronJob
+	WarehouseUploadTrigerChan chan warehouseutils.WarehouseT
+)
+
+func init() {
+	cronParser = cron.NewParser(cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor)
+	warehouseUploadCronJobMap = make(map[string]warehouseUploadCronJob)
+	WarehouseUploadTrigerChan = make(chan warehouseutils.WarehouseT)
+}
+
+func IsCronExpressionPresent(cronExpression string, warehouseIdentifier string) bool {
+	stopExistingCronFromRunningInFuture(warehouseIdentifier)
+	if len(strings.TrimSpace(cronExpression)) > 0 {
+		return true
+	}
+	return false
+}
+
+func stopExistingCronFromRunningInFuture(warehouseIdentifier string) {
+	if uploadCronJob, ok := warehouseUploadCronJobMap[warehouseIdentifier]; ok {
+		uploadCronJob.warehouseUploadCron.Remove(uploadCronJob.jobEntryID)
+	}
+}
+
+func CreateCron(cronExpression string, warehouseIdentifier string, warehouse warehouseutils.WarehouseT) error {
```

Comment:
Do we need both warehouseIdentifier and warehouse? 
Seems like warehouse alone is sufficient.